tls/s2n_handshake_io.c

```diff
@@ -1043,12 +1043,15 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
 
     if (conn->config->use_tickets) {
         if (conn->session_ticket_status == S2N_DECRYPT_TICKET) {
+            /* If ticket decryption is successful in TLS12, we can exit. TLS13
+             * decrypts session tickets in its extension processing logic. */
```

Comment:
I think we can make this a little more helpful. The confusing part isn't this line about decryption, it's how this line interacts with setting FULL_HANDSHAKE and WITH_SESSION_TICKET.
```suggestion
            /* If a valid TLS1.2 session ticket is provided, we reuse that session.
             * Otherwise, we will perform a full handshake and then generate
             * a new session ticket.
             */
```